app/liquid/liquid_i18n.rb

```diff
@@ -17,14 +31,38 @@
 # class subclassed directly on Exception that will not be caught.
 
 class I18n::TranslationMissing < Exception; end
+class I18n::TooMuchInterpolation < StandardError; end
+
+module LiquidI18n
+  MAX_INTERPOLATIONS = 10
 
-module LiquidI18nRails
-  def t(string)
-    return I18n.t(string) unless Rails.env.development? || Rails.env.test?
+  def t(query)
+    translation_key, interpolation_vals = parse_interpolation(query)
+    return I18n.t(translation_key, **interpolation_vals) unless Rails.env.development? || Rails.env.test?
     begin
-      I18n.t(string, :raise => true)
+      I18n.t(translation_key, **interpolation_vals.merge(:raise => true))
```

Comment:
Looking nice!

I can't see us having a `Rails.env.staging` environment. To me it doesn't make sense to not have staging as production., so you could DRY this a tiny bit:

```ruby
begin
  I18n.t(translation_key, **interpolation_vals.merge(raise: Rails.env.production? ))
...
```